### PR TITLE
Handle deleted migration classes #795

### PR DESCRIPTION
**Pull request by jschaedl**
|      Q       |   A
|------------- | -----------
| Type         | feature
| BC Break     | no
| Fixed issues | #795

### CODE REVIEW

**Comment by jschaedl** [comment]
> Looks good. Can you move that delete functionality to a service and add a unit test for it?

Sure I can do that. But not sure if we need a whole new Service class like `MigrationVersionRemover` 🤔

As an alternative we could add a new method to the `MigrationRepository` and name it `deleteMigrationVersionFromDatabase(string $version): void`.

WDYT?

**Comment by jwage** [comment]
MigrationRepository sounds good.

File: lib/Doctrine/Migrations/Tools/Console/Command/VersionCommand.php [hunk 1]
```diff
@@ -168,10 +168,29 @@ private function markVersions(InputInterface $input, OutputInterface $output) :
      * @throws VersionDoesNotExist
      * @throws UnknownMigrationVersion
      */
-    private function mark(OutputInterface $output, string $version, bool $all = false) : void
+    private function mark(InputInterface $input, OutputInterface $output, string $version, bool $all = false) : void
     {
         if (! $this->migrationRepository->hasVersion($version)) {
-            throw UnknownMigrationVersion::new($version);
+            if ((bool) $input->getOption('delete') === false) {
```
Comment (jwage) [inline lib/Doctrine/Migrations/Tools/Console/Command/VersionCommand.php]: I'm not sure if VersionCommand already has a test or not but we need to test this functionality/logic here too.

**Comment by jwage** [comment]
Looks good. Can you squash? and then I think we are ready to merge.

**Comment by jschaedl** [comment]
@jwage I squashed, so ready to merge :-)

**Comment by jwage** [comment]
Looks like CS is failing. Run vendor/bin/phpcbf to fix.

**Comment by jschaedl** [comment]
@jwage I think we are ready to merge. :-)

**Comment by jwage** [comment]
Thanks for your work on this!

**Pull request merged**